hoomd_benchmarks/__main__.py

```diff
@@ -45,18 +46,25 @@
                     'CSV file.')
 parser.add_argument('--name',
                     type=str,
-                    help='Name identifying this benchmark run.')
+                    default=None,
+                    help='Name identifying this benchmark run'
+                    ' (leave unset to use the HOOMD-blue version).')
 args = parser.parse_args()
 
-benchmark_args = copy.deepcopy(vars(args))
-del benchmark_args['benchmarks']
-del benchmark_args['output']
-del benchmark_args['name']
-benchmark_args['device'] = common.make_hoomd_device(args)
+benchmark_args_ref = copy.deepcopy(vars(args))
+del benchmark_args_ref['benchmarks']
+del benchmark_args_ref['output']
+del benchmark_args_ref['name']
+benchmark_args_ref['device'] = common.make_hoomd_device(args)
 
 performance = {}
 
 for benchmark_class in benchmark_classes:
+    # scale the benchmark_steps by the class specific scale factor
+    benchmark_args = copy.copy(benchmark_args_ref)
+    benchmark_args['warmup_steps'] *= benchmark_class.SUITE_STEP_SCALE
+    benchmark_args['benchmark_steps'] *= benchmark_class.SUITE_STEP_SCALE
```

Comment:
Why not just define the absolute number of steps each benchmark needs to run for warmup and benchmark in their classes, instead of using scale factors?

Reply:
With the scale factors, users can run the full suite with a non default number of base steps: `python3 -m hoomd_benchmarks --device GPU --benchmark_steps=100000` to run a longer or shorter benchmark as desired.